packages/commonwealth/client/scripts/views/pages/communities.tsx

```diff
@@ -177,12 +175,18 @@ const CommunitiesPage = () => {
         return threadCountB - threadCountA;
       })
       .map((community: CommunityInfo, i) => {
+        // allow user to buy stake if they have a connected address that matches this community's base chain
+        const canBuyStake = !!app?.user?.addresses?.find?.(
+          (address) => address?.community?.base === community?.base,
+        );
+
         return (
           <CWRelatedCommunityCard
             key={i}
             community={community}
             memberCount={community.addressCount}
             threadCount={community.threadCount}
+            canBuyStake={canBuyStake}
```

Comment:
I noticed this is an async function (canBuyStake). The eslint diff plugin is off so make be careful that it's just not a dangling promise. You have to call .catch() on it.

Reply:
not sure what you mean here? are you referring to `onStakeBtnClick`?

Reply:
canBuyStake is an async function so if you pass that to something that just calls the function without a .catch() (usually a onClick handler) then that will ignore error handling.  

A work around if it's just a onClick handler is to create a new callback which at least calls console.error

We need a UI error handler though.